### PR TITLE
CICS SEND Command Implementation

### PR DESCRIPTION
**Pull request by ilidio-lopes**
## How Has This Been Tested?

- [x] Unit tests CICS Send parser rules

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        checkTextNoedit((CICSParser.Cics_send_text_noeditContext) ctx);
+        break;
+      default:
+        break;
+    }
+    checkDuplicates(ctx);
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkGroup1(CICSParser.Cics_send_group1Context ctx) {
```
Comment (KutluOzel-b) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java:152]: I think better to add missing message for LENGTH or FLENGTH requires FROM in all groups

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    checkHasMutuallyExclusiveOptions("CNOTCOMPL or DEFRESP", ctx.CNOTCOMPL(), ctx.DEFRESP());
+    checkHasMutuallyExclusiveOptions("LDC or FMH", ctx.LDC(), ctx.FMH());
+    checkHasMutuallyExclusiveOptions("PASSBK or CBUFF", ctx.PASSBK(), ctx.CBUFF());
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkControl(CICSParser.Cics_send_control_mapContext ctx) {
+    checkHasExactlyOneOption("CONTROL or MAP", ctx, ctx.CONTROL(), ctx.MAP());
+    checkHasMutuallyExclusiveOptions("ERASE or ERASEAUP", ctx.ERASE(), ctx.ERASEAUP());
+    if (!ctx.DEFAULT().isEmpty() || !ctx.ALTERNATE().isEmpty()) {
```
Comment (KutluOzel-b) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java]: Although it is stated as this is docs, ERASE is not mandatory to use DEFAULT or ALTERNATE 
image

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+  @SuppressWarnings("unchecked")
+  private void checkGroup1(CICSParser.Cics_send_group1Context ctx) {
+    checkHasMutuallyExclusiveOptions("LENGTH or FLENGTH", ctx.LENGTH(), ctx.FLENGTH());
+    checkHasMutuallyExclusiveOptions("INVITE or LAST", ctx.INVITE(), ctx.LAST());
+    checkHasMutuallyExclusiveOptions("STRFIELD or ERASE", ctx.STRFIELD(), ctx.ERASE());
+    checkHasMutuallyExclusiveOptions("DEFAULT or ALTERNATE", ctx.DEFAULT(), ctx.ALTERNATE());
+    if (!ctx.DEFAULT().isEmpty() || !ctx.ALTERNATE().isEmpty()) {
+      checkHasMandatoryOptions(ctx.ERASE(), ctx, "ERASE");
+    }
+    checkHasMutuallyExclusiveOptions("STRFIELD or CTLCHAR", ctx.STRFIELD(), ctx.CTLCHAR());
```
Comment (KutluOzel-b) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java:160]: ATTACHID also conflicts with CTLCHAR
image

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    checkHasMutuallyExclusiveOptions("CONFIRM or WAIT", ctx.CONFIRM(), ctx.WAIT());
+    checkHasMutuallyExclusiveOptions("CNOTCOMPL or DEFRESP", ctx.CNOTCOMPL(), ctx.DEFRESP());
+    checkHasMutuallyExclusiveOptions("LDC or FMH", ctx.LDC(), ctx.FMH());
+    checkHasMutuallyExclusiveOptions("PASSBK or CBUFF", ctx.PASSBK(), ctx.CBUFF());
```
Comment (KutluOzel-b) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java]: PASSBK CBUFF and CNOTCOMPL are mutual exclusive 
image

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java (hunk 1 above)
Comment (KutluOzel-b) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java:152]: FROM is required when FMH is in use
image

**Review by KutluOzel-b — CHANGES_REQUESTED** [review]

**Comment by ilidio-lopes** [comment]
All comments addressed in the last commit

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckUtility.java [hunk 5]
```diff
@@ -66,12 +66,12 @@ public CICSOptionsCheckUtility(DialectProcessingContext context, List<SyntaxErro
     optionsMap.put(
         CICSWaitOptionsCheckUtility.RULE_INDEX,
         new CICSWaitOptionsCheckUtility(context, errors));
-    optionsMap.put(
-        CICSAssignOptionsCheckUtility.RULE_INDEX,
```
Comment (KutluOzel-b) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckUtility.java]: Its good to keep this :)

Comment (ilidio-lopes) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckUtility.java]: it is :)

Comment (ilidio-lopes) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckUtility.java]: fix pushed

**Review by KutluOzel-b — APPROVED** [review]
Looks good

**Comment by amanPras** [comment]
@KutluOzel-b Thanks for the review

**Pull request merged**